### PR TITLE
fix(Core): Fixed crash happening when clearing old mail

### PR DESCRIPTION
**Pull request by peti446**
## CHANGES PROPOSED:
- Clear the array correctly
- Updated total mail variable handling to avoid false positives on system mail

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3603

## TESTS PERFORMED:
- Create new character level it from 1-80 using .level 79 check if server 
- Created new character with instant 80
- Send mail from console
- Check if any action crashed the server

## HOW TO TEST THE CHANGES:

- Create new character level it from 1-80 using .level 79 check if server 
- Created new character with instant 80
- Send mail from console
- Check if any action crashed the server

## KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -20093,6 +20094,8 @@ void Player::_SaveMail(SQLTransaction& trans)
             stmt = CharacterDatabase.GetPreparedStatement(CHAR_DEL_MAIL_ITEM_BY_ID);
             stmt->setUInt32(0, m->messageID);
             trans->Append(stmt);
+            if(totalMailCount > 0)
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Player/Player.cpp:20097]: missing space after the `if`

**Comment by FrancescoBorzi** [comment]
@peti446 thanks for quickly providing a solution to the crash!

**Review by Si1ker — APPROVED** [review]

**Comment by pangolp** [comment]
AzerothCore rev. d027239193dd 2020-10-23 18:49:31 -0300 (**pr-3604 branch**) (Win64, RelWithDebInfo) (worldserver-daemon) ready...

I just tested it on Windows 10. Before PR, it crashed at 80. Thanks @peti446. Excellent work.

**Review by pangolp — APPROVED** [review]

**Comment by Crazy-Mathematical** [comment]
Cool man, thanks for all.

**Comment by Koschull** [comment]
Thanks so much for quick repair :) Works well on W10 64Bit.

**Comment by Ramys** [comment]
Thanks for all.

**Pull request merged**